client/src/dataset/sagas.ts

```diff
@@ -63,12 +63,19 @@ export function* openDatasetSagaFullPath(action: ReturnType<typeof browserAction
 export function* openDatasetSaga(action: ReturnType<typeof browserActions.Actions.select>) {
     const config: ConfigState = yield select((state: RootReducer) => state.config);
     const fullPath = joinPaths(config, action.payload.path, action.payload.name);
-    yield call(doOpenDataset, fullPath);
+    if(fullPath.includes(".dm") || fullPath.includes(".DM")) {
+      const timestamp = Date.now();
+      const id = uuid();
+      yield put(datasetActions.Actions.detectFailed(fullPath));
+      yield put(datasetActions.Actions.error(`DM`, `DM dataset is currently not supported in the GUI`, timestamp, id));
+    } else {
```

Comment:
Nice, this should work well for .dm files! Though instead of `includes`, I would check the end of the string for dm3 or dm4, as a file in another format may very well be called `frame-54.dmsomething-123.bin`, for example, which would be caught by your above check.

As an improvement, it should be possible to do this in a more general way: all GUI-supported formats are listed in the `DatasetTypes` enum, so if the detect step returns a type that is not part of that enum, the above error could be triggered. This may need some "convincing" the type system, but should be possible. Let me know if you need help at any time!

If we have this more general way, if will work well even in the face of changes: if, some day soon (maybe as part of #498), we start supporting DM, the check doesn't need to be changed; also, if we later add a new format to the backend and don't support it from the beginning in the GUI, the same check will still work.

Reply:
> If we have this more general way, if will work well even in the face of changes

Thanks!